packages/shared/src/mock-utils.ts

```diff
@@ -1,118 +1,52 @@
-import { buildEntityPath } from '../hdf5-utils';
-import type { SilxStyle } from '../nexus-models';
-import type { MockAttribute, MockDataset, MockValueId } from './models';
-import { mockValues } from './values';
-
-/* ----------------- */
-/* ----- TYPES ----- */
-
-export function intType(
-  size: 8 | 16 | 32 | 64 = 32,
-  unsigned = false,
-  endianness = Endianness.LE,
-): NumericType {
-  return {
-    class: unsigned ? DTypeClass.Unsigned : DTypeClass.Integer,
-    endianness,
-    size,
-  };
-}
-
-export function floatType(
-  size: 16 | 32 | 64 = 32,
-  endianness = Endianness.LE,
-): NumericType {
-  return { class: DTypeClass.Float, endianness, size };
-}
-
-export function strType(
-  charSet: StringType['charSet'] = 'ASCII',
-  length?: number,
-): StringType {
-  return {
-    class: DTypeClass.String,
-    charSet,
-    ...(length !== undefined && { length }),
-  };
-}
-
-export function boolType(): BooleanType {
-  return { class: DTypeClass.Bool };
-}
-
-export function cplxType(
-  realType: NumericType,
-  imagType = realType,
-): ComplexType {
-  return { class: DTypeClass.Complex, realType, imagType };
-}
-
-export function compoundType(fields: Record<string, DType>): CompoundType {
-  return { class: DTypeClass.Compound, fields };
-}
-
-export function printableCompoundType(
-  fields: Record<string, PrintableType>,
-): PrintableCompoundType {
-  return { class: DTypeClass.Compound, fields };
-}
-
-export function unknownType(): UnknownType {
-  return { class: DTypeClass.Unknown };
-}
-
-function guessType(value: unknown): DType {
-  if (typeof value === 'number') {
-    return floatType(64);
-  }
-
-  if (typeof value === 'string') {
-    return strType();
-  }
-
-  if (typeof value === 'boolean') {
-    return boolType();
+} from './hdf5-models';
+import { EntityKind } from './hdf5-models';
+import {
+  boolType,
+  buildEntityPath,
+  cplxType,
+  floatType,
+  strType,
+  unknownType,
+} from './hdf5-utils';
+import type { MockAttribute, MockDataset, MockValueId } from './mock-models';
+import { mockValues } from './mock-values';
+import type { SilxStyle } from './nexus-models';
+
+export function assertMockDataset<S extends Shape, T extends DType>(
+  dtst: Dataset<S, T>,
+): asserts dtst is MockDataset<S, T> {
+  if (!('value' in dtst)) {
+    throw new Error('Expected mock dataset');
   }
+}
 
-  if (
-    Array.isArray(value) &&
-    value.length === 2 &&
-    typeof value[0] === 'number'
-  ) {
-    return cplxType(floatType(64));
+export function assertMockAttribute<S extends Shape, T extends DType>(
+  attr: Attribute<S, T>,
+): asserts attr is MockAttribute<S, T> {
+  if (!('value' in attr)) {
+    throw new Error('Expected mock attribute');
   }
-
-  return unknownType();
 }
 
 /* ---------------------- */
 /* ----- ATTRIBUTES ----- */
 
-export function attr<S extends Shape, T extends DType>(
+export function attribute<S extends Shape, T extends DType>(
```

Comment:
Using the full word for the base utility, in line with `dataset()`, `datatype()` and `group()`, which allows me to use the abbreviation for a variable above.